File: 03-poo/10-miembros-de-clase-y-de-instancia/java/ejemplos/ClaseEInstancia.java
public class ClaseEInstancia {
    static class Usuario {
        private static int creados;
        private final String nombre;
        Usuario(String nombre){ this.nombre=nombre; creados++; }
        static int getCreados(){ return creados; }
        String getNombre(){ return nombre; }
    }
    public static void main(String[] args){ new Usuario("Ana"); new Usuario("Luis"); System.out.println(Usuario.getCreados()); }
}

File: 03-poo/10-miembros-de-clase-y-de-instancia/scala/ejemplos/ClaseEInstancia.scala
class Usuario(val nombre: String):
  Usuario.registrar()
object Usuario:
  private var creados = 0
  private def registrar(): Unit = creados += 1
  def total: Int = creados

@main def claseEInstancia(): Unit =
  new Usuario("Ana"); new Usuario("Luis")
  println(Usuario.total)
